main.go

```diff
@@ -103,6 +138,45 @@ func triggerShellyUpdateCheck(hostname string) (*shellyUpdateCheckResponse, erro
 	return checkStatus, nil
 }
 
+func makeShellyInfoRequest(hostname string) (*shellyInfoResponse, error) {
```

Comment:
I suggest adding `Gen2` in the name of the new methods to differentiate them from the old ones.

```suggestion
func makeShellyGen2InfoRequest(hostname string) (*shellyInfoResponse, error) {
```

Reply:
This entry point is actually common for gen1 and gen2. However, I've removed it from the PR because it was unused. I had used it in an earlier version to determine the generation but then found that I could already get that information from the service TXT records.